core/src/executor/evaluate/expr.rs

```diff
@@ -23,6 +23,10 @@ pub fn binary_op<'a>(
     l: Evaluated<'a>,
     r: Evaluated<'a>,
 ) -> Result<Evaluated<'a>> {
+    if l.is_null() || r.is_null() {
+        return Ok(Evaluated::null());
+    }
```

Comment:
it would be redundant to the below logic.
`evaluate_eq` is supposed to return `NULL` if the either left or right is `NULL` so we may not need this.

Reply:
The same would need to be added to all the operations below, not only evaluate_eq. This way we avoid changing all of those operations. If you prefer, I can change all of the below operations to return `Nullable`, but I thought you preferred minimal changes.

Reply:
this pr is targeting only `=` evaluate_eq operations. if something other also needs to be fixed, then we can do it in the following pr. let's focus only for the `evaluate_eq` for now.

Reply:
This PR changes that operation by propagating the NULL. At this time, the implementation of the other operations, like all `binary_op` stops the propagation of the NULL. Either those operations also propagate the NULL, or it stops there. In order to propagate the NULL while making minimal changes, I have added that line.

Reply:
ok, I understand the goal what we do for this and following PRs are for making the NULL propagation complete.
however, for us to do efficient review, the big single task needs to be splitted into small steps. that is why we focus only on `evaluate_eq` in this PR.
if we also want to handle other `binary_op` cases, this PR needs to be quite bigger than now.
could we limit the scope of this PR to make `NULL = {value}` or `{value} = NULL` return `NULL` which was previously `False`?

Reply:
I will try to change that, though I suspect tests may fail. Hopefully, I am wrong on this and it is as easy as that.

Reply:
I was mistaken - I suppose we do not have tests at this time that expect propagation across AND, OR and other such operations.  See commit [68d9f87](https://github.com/gluesql/gluesql/pull/1577/commits/68d9f872e73b1660e75a024e68eeb79f470ff546)